From 7b01aef5046045208d2fc3e2508a10be3ec714d1 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Mon, 20 Jun 2022 18:32:23 +0300
Subject: [PATCH] SW-70527] ras: add memory for sram_bist data

Added memory details in memory framework for the
sram_bist temporary data storage. This is used by
ppboot to store the sram bist result details.
Later preboot would read this info and write into boot_info.

Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
Change-Id: I6ebe87e819c4bd85116895c70d762a1bd56a496d
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index a2ced03..d169794 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -25,6 +25,7 @@
 #include "common.h"
 #include <hl.h>
 #include <errno.h>
+#include <memory.h>
 
 #define SDRAM_BASE_FULL 0x1000FFFFFD000000ull
 
diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 7ee9f12..ffa8cac 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -83,6 +83,9 @@
 #define PPBOOT_DCCM_BASE	0x27064030
 #define PPBOOT_DCCM_SZ		0x2000	/* 8KB */
 
+#define SRAM_BIST_DATA_BASE	(PPBOOT_DCCM_BASE + PPBOOT_DCCM_SZ)
+#define SRAM_BIST_DATA_SZ	0x400	/* 1KB */
+
 #define HL_SPI_BASE		0x26000000
 #define HL_CFG_BASE		0xF0000000
 #define I2C_SLAVE_DEFAULT_INTR_MASK	0x48FF
diff --git a/include/memory.h b/include/memory.h
index 9622de9..5065555 100644
--- a/include/memory.h
+++ b/include/memory.h
@@ -20,11 +20,24 @@
 #define MEM_I2C_BASE		(SDRAM_BASE + 0x800000)	/* SRAM_BASE + 8M */
 #define MEM_I2C_SZ		0x200000		/* 2M */
 
+#if defined(SRAM_BIST_DATA_BASE)
+#define MEM_SRAM_BIST_DATA_BASE	SRAM_BIST_DATA_BASE	/* Addr to store SRAM BIST data in ppboot */
+#else
+#define MEM_SRAM_BIST_DATA_BASE	0			/* Dummy value to get compilation through */
+#endif
+
+#if defined(SRAM_BIST_DATA_SZ)
+#define MEM_SRAM_BIST_DATA_SZ	SRAM_BIST_DATA_SZ	/* Size of SRAM BIST data */
+#else
+#define MEM_SRAM_BIST_DATA_SZ	0			/* Dummy value to get compilation through */
+#endif
+
 enum mem_region {
 	MEM_GP,
 	MEM_LKD_PCI,
 	MEM_FWUPD_PCI,
 	MEM_I2C,
+	MEM_SRAM_BIST,
 	MEM_NOENT	/* terminator */
 };
 
diff --git a/lib/preboot/memory.c b/lib/preboot/memory.c
index 793f2b1..c917dca 100644
--- a/lib/preboot/memory.c
+++ b/lib/preboot/memory.c
@@ -29,6 +29,10 @@
 		.base = MEM_I2C_BASE,
 		.size = MEM_I2C_SZ,
 	},
+	[MEM_SRAM_BIST] = {
+		.base = MEM_SRAM_BIST_DATA_BASE,
+		.size = MEM_SRAM_BIST_DATA_SZ,
+	},
 };
 
 __weak int preboot_get_dev_spec_mem_region(enum mem_region region,
@@ -46,6 +50,7 @@
 	case MEM_LKD_PCI:
 	case MEM_FWUPD_PCI:
 	case MEM_I2C:
+	case MEM_SRAM_BIST:
 		break;
 	default:
 		return 1;
